libwild/src/elf.rs

```diff
@@ -355,18 +355,14 @@ pub(crate) const COMPRESSION_HEADER_SIZE: usize =
     size_of::<object::elf::CompressionHeader64<LittleEndian>>();
 
 pub(crate) const GOT_ENTRY_SIZE: u64 = 0x8;
-pub(crate) const PLT_ENTRY_SIZE: u64 = PLT_ENTRY_TEMPLATE.len() as u64;
+// TODO: Right now, both x86_64 and AArch64 have 16 byte long entires, but
```

Comment:
One way this could be handled is by adding an `element_size_fn` that could dynamically compute the element size. This would be similar to what's currently done with `info_fn`.

Another options would be a more drastic refactoring to make the whole `SECTION_DEFINITIONS` be computed dynamically. I'm not sure what the implications of that would be though.

Reply:
Thanks for the ideas! I've briefly tried the `info_fn` path and it seems one needs to add the `Arch` trait into `BuiltInSectionDetails` anyway (due to the `info_fn` member).